### PR TITLE
[#40] Server type filter

### PR DESCRIPTION
**Pull request by xRay2016**
Server type filter with `both`, `primary` and `replica` options.

### CODE REVIEW

File: doc/YAML.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Each yaml is sequcence of queries. Every query can include four parts. See a [sa
 |----------|---------|----------|-------------|
 | query | | Yes | The query sql of metric |
 | tag | | Yes | The tag of metric |
+| server_query_type  | both | No | The query on which server type. Valid options: `both`, `primary`, `replica` |
```
Comment (jesperpedersen) [inline doc/YAML.md:14]: Let just call it `server`

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: doc/YAML.md (hunk 1 above)
Comment (jesperpedersen) [reply doc/YAML.md:14]: Still here

File: src/include/pgexporter.h [hunk 2]
```diff
@@ -161,6 +165,7 @@ struct prometheus
    char query[MAX_QUERY_LENGTH];                   /*< The query string of metric */
    char tag[MISC_LENGTH];                          /*< The metric name */
    int sort_type;                                  /*< Sorting type of multi queries 0--SORT_NAME 1--SORT_DATA0 */
+   int server;                          /*< Query type 0--SERVER_QUERY_BOTH 1--SERVER_QUERY_PRIMARY 2--SERVER_QUERY_REPLICA */
```
Comment (jesperpedersen) [inline src/include/pgexporter.h:168]: You can call the internal data type `server_query_type` if you want

Comment (jesperpedersen) [reply src/include/pgexporter.h:168]: Either way

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Small change - can you make the modification and force push ?

File: src/libpgexporter/yaml_configuration.c [hunk 3]
```diff
@@ -332,6 +332,26 @@ handle_key_value(struct prometheus* metric, char* key, char* value)
...
+      {
+         metric->server_query_type = SERVER_QUERY_PRIMARY;
+      }
+      else if (!strcmp(value, "replica"))
+      {
+         metric->server_query_type = SERVER_QUERY_REPLICA;
+      }
+      else
+      {
+         pgexporter_log_error("pgexporter: unexpected server_query_type %s", value);
```
Comment (jesperpedersen) [inline src/libpgexporter/yaml_configuration.c:351]: Just user-facing one left.

Squash and force push

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: src/libpgexporter/yaml_configuration.c (hunk 3 above)
Comment (xRay2016) [reply src/libpgexporter/yaml_configuration.c]: It is OK now.

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**